crates/uv/src/commands/project/mod.rs

```diff
@@ -149,6 +149,15 @@ pub(crate) enum ProjectError {
     #[error("Default group `{0}` (from `tool.uv.default-groups`) is not defined in the project's `dependency-group` table")]
     MissingDefaultGroup(GroupName),
 
+    #[error("Extra `{0}` is not defined in the project's `optional-dependencies` table")]
+    MissingExtraProject(ExtraName),
+
+    #[error("Extra `{0}` is not defined in any project's `optional-dependencies` table")]
+    MissingExtraWorkspace(ExtraName),
```

Comment:
Wording could probably be adapted here, as the check happens at lock file level, but not sure exactly how we would want to phrase this.